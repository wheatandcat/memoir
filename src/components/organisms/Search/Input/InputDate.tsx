import React, { memo, useState } from 'react';
import { StyleSheet, TouchableOpacity, ViewStyle } from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import DateTimePickerModal from 'react-native-modal-datetime-picker';
import dayjs from 'lib/dayjs';

export type Props = {
  startDate: Date;
  endDate: Date;
  error?: boolean;
  onChangeStartDate: (date: Date) => void;
  onChangeEndDate: (date: Date) => void;
};

const InputDate: React.FC<Props> = (props) => {
  const [openStartDate, setOpenStartDate] = useState(false);
  const [openEndDate, setOpenEndDate] = useState(false);

  const dateText: ViewStyle[] = [styles.dateText];

  if (props.error) {
    dateText.push(styles.errorDateText);
  }

  return (
    <View pt={3} style={styles.inputDate}>
      <View style={dateText}>
        <TouchableOpacity onPress={() => setOpenStartDate(true)}>
          <Text variants="middle" color={props.error ? 'error' : 'secondary'}>
            {dayjs(props.startDate).format('YYYY.MM.DD')}
          </Text>
        </TouchableOpacity>
        <DateTimePickerModal
          isVisible={openStartDate}
          mode="date"
          locale="ja"
          confirmTextIOS="完了"
          cancelTextIOS="キャンセル"
          is24Hour={true}
          date={props.startDate}
          onConfirm={(v) => {
            setOpenStartDate(false);
            props.onChangeStartDate(v);
          }}
          onCancel={() => setOpenStartDate(false)}
        />
      </View>
      <View mx={2}>
        <Text color={props.error ? 'error' : 'secondary'}>-</Text>
      </View>
      <View style={dateText}>
        <TouchableOpacity onPress={() => setOpenEndDate(true)}>
          <Text variants="middle" color={props.error ? 'error' : 'secondary'}>
            {dayjs(props.endDate).format('YYYY.MM.DD')}
          </Text>
        </TouchableOpacity>
        <DateTimePickerModal
          isVisible={openEndDate}
          mode="date"
          locale="ja"
          confirmTextIOS="完了"
          cancelTextIOS="キャンセル"
          is24Hour={true}
          date={props.endDate}
          onConfirm={(v) => {
            setOpenEndDate(false);
            props.onChangeEndDate(v);
          }}
          onCancel={() => setOpenEndDate(false)}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  inputDate: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  dateText: {
    borderColor: theme().color.secondary.main,
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  errorDateText: {
    borderColor: theme().color.error.main,
    borderBottomWidth: 1,
  },
});

export default memo(InputDate);
