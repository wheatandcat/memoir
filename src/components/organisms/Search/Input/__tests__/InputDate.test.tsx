import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import InputDate, { Props } from '../InputDate';

const propsData = (): Props => ({});

describe('components/organisms/Search/Input/InputDate.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<InputDate {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
