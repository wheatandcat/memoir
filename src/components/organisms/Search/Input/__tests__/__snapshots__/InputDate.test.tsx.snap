// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/Search/Input/InputDate.tsx 正常にrenderすること 1`] = `
<View
  pt={3}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "#362740",
        },
      ]
    }
  >
    <TouchableOpacity
      onPress={[Function]}
    >
      <Text
        color="secondary"
        fontFamily="RobotoCondensed-Bold"
        variants="middle"
      >
        2021.01.01
      </Text>
    </TouchableOpacity>
    <DateTimePickerModal
      backdropStyleIOS={Object {}}
      cancelTextIOS="キャンセル"
      confirmTextIOS="完了"
      date={2020-12-31T15:00:00.000Z}
      is24Hour={true}
      isVisible={false}
      locale="ja"
      modalPropsIOS={Object {}}
      mode="date"
      onCancel={[Function]}
      onConfirm={[Function]}
      pickerContainerStyleIOS={Object {}}
      pickerStyleIOS={Object {}}
    />
  </View>
  <View
    mx={2}
  >
    <Text
      color="secondary"
      fontFamily="RobotoCondensed-Bold"
      variants="body"
    >
      -
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "#362740",
        },
      ]
    }
  >
    <TouchableOpacity
      onPress={[Function]}
    >
      <Text
        color="secondary"
        fontFamily="RobotoCondensed-Bold"
        variants="middle"
      >
        2021.01.18
      </Text>
    </TouchableOpacity>
    <DateTimePickerModal
      backdropStyleIOS={Object {}}
      cancelTextIOS="キャンセル"
      confirmTextIOS="完了"
      date={2021-01-17T15:00:00.000Z}
      is24Hour={true}
      isVisible={false}
      locale="ja"
      modalPropsIOS={Object {}}
      mode="date"
      onCancel={[Function]}
      onConfirm={[Function]}
      pickerContainerStyleIOS={Object {}}
      pickerStyleIOS={Object {}}
    />
  </View>
</View>
`;
